api/hypershift/v1beta1/hostedcluster_types.go

```diff
@@ -1725,17 +1725,27 @@ type AzurePlatformSpec struct {
 
 	// ResourceGroupName is the name of an existing resource group where all cloud resources created by the Hosted
 	// Cluster are to be placed.
+	// Resource group naming requirements can be found here: https://azure.github.io/PSRule.Rules.Azure/en/rules/Azure.ResourceGroup.Name/.
 	// In ARO HCP, this will be the managed resource group where customer cloud resources will be created.
 	//
 	// Example: if your resource group ID is /subscriptions/<subscriptionID>/resourceGroups/<resourceGroupName>, your
 	//          ResourceGroupName is <resourceGroupName>.
 	//
 	// +kubebuilder:default:=default
 	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^[a-zA-Z0-9_()\-\.]{1,89}[a-zA-Z0-9_()\-]$`
 	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceGroupName is immutable"
 	// +immutable
 	ResourceGroupName string `json:"resourceGroup"`
 
+	// NetworkResourceGroupName is the name of an existing resource group provided by a client; it is expected to contain
+	// an existing, pre-setup VNET, Subnet and NSG. This field is only used by ARO HCP.
+	//
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf", message="NetworkResourceGroupName is immutable"
+	// +optional
+	// +immutable
+	NetworkResourceGroupName string `json:"networkResourceGroupName,omitempty"`
```

Comment:
Is this name pending some renaming depending on https://redhat-internal.slack.com/archives/C05Q232L29E/p1714763029977549?thread_ts=1714659596.338989&cid=C05Q232L29E?